### PR TITLE
test/e2e: More refactoring

### PR DESCRIPTION
**Pull request by cgwalters**
Rather than poll all of the daemons, add a helper that waits
for a pool to complete a config.

One of our tests walks over the MCDs, change it to just assert
on all of the nodes.

The SSH test can also just wait for a pool and then `rsh` to
each node.

### CODE REVIEW

File: test/e2e/mcd_test.go [hunk 1]
```diff
@@ -136,28 +157,16 @@ func TestMCDeployed(t *testing.T) {
...
-					if len(visited) == len(nodes) {
-						return true, nil
-					}
-					continue
-				}
-			}
-			return false, nil
-		}); err != nil {
-			t.Errorf("machine config didn't result in file being on any worker: %v", err)
+		if err := waitForPoolComplete(cs, "worker", newMCName); err != nil {
```
Comment (runcom) [inline test/e2e/mcd_test.go]: isn't thi racy if we jump here before the pool starts rolling? I believe that's why I went node by node confirming current is there and of the new pool

Comment (runcom) [reply test/e2e/mcd_test.go]: ok, you added a check below but this may flake then?

Comment (cgwalters) [reply test/e2e/mcd_test.go]: Just above that we do `waitForRenderedConfig` right?

Comment (cgwalters) [reply test/e2e/mcd_test.go]: For the record though I didn't test this much locally, threw it up to have CI do it.

Comment (runcom) [reply test/e2e/mcd_test.go]: oh yeah, I had an old master on my laptop and didn't notice that, cool yeah, no flake no race :tada:

**Comment by cgwalters** [comment]
Hum, so that run failed; added some more debug logs to help me figure it out.  Offhand...I think there *is* a race here because the tests were relying on checking for config=X and status=Updated, but we don't update those atomically.  Really...it feels like we should have `Spec.Configuration` and `Status.Configuration` [just like other objects](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#spec-and-status) right?

**Comment by kikisdeliveryservice** [comment]
would checking if state is Done before checking that they match clarify what's happening  (ie switch the 2 asserts?)

**Comment by cgwalters** [comment]
OK this test passes "locally" now - I added a `time.Sleep(13*time.Second)` as a temporary hack.  Pretty sure the race is https://github.com/openshift/machine-config-operator/pull/765#issuecomment-493136113 - going to take a quick look at a PR for that.

**Comment by cgwalters** [comment]
Hooray, e2e-aws-op passed!

https://github.com/openshift/machine-config-operator/pull/773 should help us avoid the race.

**Comment by kikisdeliveryservice** [comment]
ive seen these weird e2e-aws-upgrade failures before, checking them out elsewhere. let's try again:
/test e2e-aws-upgrade

**Comment by cgwalters** [comment]
Rebased :surfing_man: and CI is good, can I get a lgtm?

**Pull request merged**